src/NetCoreInstaller/Targets/packaging/windows/NetCoreInstaller/generatenetcoreinstaller.ps1

```diff
@@ -0,0 +1,186 @@
+    [Parameter(Mandatory=$true)][string]$NetStandardTargetingPackX64InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$NetStandardTargetingPackX86InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$SdkToolsetX64InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$SdkToolsetX86InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$SharedFrameworkX64InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$SharedFrameworkX86InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$SharedHostX64InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$SharedHostX86InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$WindowsDesktopTargetingPackX64InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$WindowsDesktopTargetingPackX86InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$WinFormsAndWpfSharedFrameworkX64InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$WinFormsAndWpfSharedFrameworkX86InstallerFileName,
+    [Parameter(Mandatory=$true)][string]$NetCoreVersion,
+    [Parameter(Mandatory=$true)][string]$AspNetCoreVersion,
+    [Parameter(Mandatory=$true)][string]$WindowsDesktopVersion,
+    [Parameter(Mandatory=$true)][string]$PermanentToolsetBaseURL,
+    [Parameter(Mandatory=$true)][string]$PermanentNetCoreBaseURL,
+    [Parameter(Mandatory=$true)][string]$PermanentAspNetCoreBaseURL,
+    [Parameter(Mandatory=$true)][string]$PermanentWindowsDesktopBaseURL,
+    [Parameter(Mandatory=$true)][string]$WixRoot,
+    [Parameter(Mandatory=$true)][string]$InstallerName,
+    [Parameter(Mandatory=$true)][string]$InstallerVersion,
+    [Parameter(Mandatory=$true)][string]$UpgradeAndProviderCode,
+    [Parameter(Mandatory=$true)][string]$SourceMSIDirectory,
+    [Parameter(Mandatory=$true)][string]$NetCoreInstallerArchitecture
+)
+
+function RunCandleForNetCoreInstaller
+{
+    $result = $true
+    pushd "$WixRoot"
+
+    Write-Information "Running candle for NetCore Installer.."
+
+    $candleOutput = .\candle.exe -nologo `
+        -dDotnetSrc="$inputDir" `
+        -dAspNetCoreSharedFrameworkX64InstallerFileName="$AspNetCoreSharedFrameworkX64InstallerFileName" `
+        -dAspNetCoreSharedFrameworkX86InstallerFileName="$AspNetCoreSharedFrameworkX86InstallerFileName" `
+        -dAspNetCoreV2ModuleX64InstallerFileName="$AspNetCoreV2ModuleX64InstallerFileName" `
+        -dAspNetCoreV2ModuleX86InstallerFileName="$AspNetCoreV2ModuleX86InstallerFileName" `
+        -dAspNetTargetingPackX64InstallerFileName="$AspNetTargetingPackX64InstallerFileName" `
+        -dAspNetTargetingPackX86InstallerFileName="$AspNetTargetingPackX86InstallerFileName" `
+        -dHostFxrX64InstallerFileName="$HostFxrX64InstallerFileName" `
+        -dHostFxrX86InstallerFileName="$HostFxrX86InstallerFileName" `
+        -dNetCoreAppHostPackX64InstallerFileName="$NetCoreAppHostPackX64InstallerFileName" `
+        -dNetCoreAppHostPackX64_ArmInstallerFileName="$NetCoreAppHostPackX64_ArmInstallerFileName" `
```

Comment:
This pattern is pretty overwhelming, having to pass in each var individually from MSBuild into this PowerShell script which then has to pass each one through individually. Maybe now isn't the right time to fix this--maybe once the WiX wrapping I wrote in Core-Setup is actually shared, that can be used--but it seems unfortunate that this pattern isn't limited to "old" installers.

Reply:
Is there a reason why you don't just directly build the .wixproj? I've always wondered why Core relies on PS to build the MSIs

Reply:
I briefly tried `wixproj` in Core-Setup, but it depends on .NET framework for its tasks and I didn't want to run an inner non-Core build. But Core-Setup doesn't depend on a PowerShell script, it uses `Exec` against the WiX tools.

Reply:
Resolving--probably not practical to take any action here during this PR. If I move Core-SDK onto shared tooling, getting away from the PowerShell will definitely happen. Until then, this is already coded up, and to be fair, it is the pattern in the repo so far.